### DISCUSSION TITLE
how can codeql include dependencies source in database when building database(maven project)

### DISCUSSION THREAD

**Discussion by bluesatchel** [github/codeql#9055] [Q&A] [answered]
I tried to use codeql to review ysoserial,i  use --command="mvn clean install --file pom.xml"to package and compile the project to generate the codeql-database,but i found i can only query the class which loaded and used in my code.There should be many classes from commons-collections that contain transform

![image](https://user-images.githubusercontent.com/77114775/167256287-72744c23-8f52-4ff1-b2b9-426f2ef1e5da.png)

![image](https://user-images.githubusercontent.com/77114775/167256266-15b96d3b-877c-483d-a7f5-714d38f717c4.png)

i want to know how to use codeql to query the third party dependencies' code

i really like the codeql, i think its potential is unlimated
my English isn't good,Anyway, thank you

**Comment by smowton** [comment, marked as answer]
If you want to analyse the code of commons-collections itself you should create a database for that project, not something that uses it. Or, you could download the ready-made one from lgtm.com: https://lgtm.com/projects/g/apache/commons-collections/ci/#ql

**Reply by bluesatchel** [reply]
tanks for your reply, that's helpful, i hope codeql could have the function to add the dependencies' code to the code-database in the future
